test.js

```diff
@@ -142,3 +142,60 @@ test('.cancel() method', async t => {
 	t.is(debounced(3), undefined);
 	t.is(count, 0);
 });
+
+test('before:false after:true `maxWait` option', async t => {
+	let count = 0;
+
+	const debounced = debounceFn(value => {
+		count++;
+		return value;
+	}, {
+		wait: 40,
+		maxWait: 50,
+		after: true,
+		before: false
+	});
+
+	t.is(debounced(1), undefined);
+	await delay(30);
+	t.is(count, 0);
+
+	t.is(debounced(2), undefined);
+	await delay(30);
+	t.is(count, 1);
+
+	t.is(debounced(3), 1);
+
+	await delay(200);
+	t.is(count, 2);
+});
+
+test('before:true after:false `maxWait` option', async t => {
```

Comment:
Should probably also test `before:true, after:true` and `before:false, after:false`?

Reply:
Ive added a `before:true, after:true` case, but I think the existing test for `before:false, after:false` covers it sufficiently as it throws instead of creating the function